src/metrics/promotions.ts

```diff
@@ -33,21 +35,62 @@ export class PromotionMetric implements Metric {
     this.totals = { q: 0, r: 0, b: 0, n: 0 };
 
     this.promotionMap = promotionMap as any;
+
+    this.movesAndGamesWithMaxQueenCount = [];
   }
 
   processGame(
     game: { move: PrettyMove; board: Piece[] }[],
     metadata?: string[]
   ) {
-    for (const { move, board } of game) {
+
+    // 2 queens to start with
+    let thisGameQueenCount = 2;
+
+    for (const { move } of game) {
       // TODO: we can use flags instead of includes('=)
 
       if (move.originalString.includes('=')) {
         // update maps
         this.promotionMap[move.uas][move.promotion]++;
+        
+        // increment queen count if a promotion occurs
+        if (move.promotion === "q") {
+          thisGameQueenCount++;
+        }
       }
 
-      for (const piece of board) {
+      // decrement queen count if queen is captured
+      if (move.capture && (move.capture.uas === "q" || move.capture.uas === "Q")) {
+        thisGameQueenCount--;
+      }
+
+      // identify the maxQueenCount in a particular move, and the games and moves that the maxQueenCount occured in
+      // push to array of games and moves if tie, otherwise wipe the array and add new game
+      // only add one move entry for each game maxQueenCount (rather than one entry for each move that the maxQueenCount appears in)
+      if (thisGameQueenCount > this.maxQueenCounts) {
+        this.maxQueenCounts = thisGameQueenCount;
+        const gameSite = metadata.find((item) => item.startsWith('[Site "'))
+          ?.replace('[Site "', '')
+          ?.replace('"]', '');
+        const moveOS = move.originalString;
+        if (!this.movesAndGamesWithMaxQueenCount.some((item) => item.game === gameSite)) {
+          this.movesAndGamesWithMaxQueenCount = [{
+            game: gameSite,
+            move: moveOS
+          }];
+        }
+      } else if (thisGameQueenCount === this.maxQueenCounts) {
```

Comment:
there are bugs with this logic. You could solve them by using a set instead. But as it is if a game goes up to the max, loses a queen then gets another you'll have the same game 2x

Reply:
I disagree. There is a check to see if the game already exists in the movesAndGamesWithMaxQueenCount array. If the game exists in that array it won't be added